packages/python/src/armonik/common/channel.py

```diff
@@ -76,6 +76,33 @@ def _load_certificates(
     return certificate_authority, client_certificate, client_key
 
 
+def get_scheme_and_endpoint_from_uri(
+    uri: str, forced_scheme: Optional[str] = None, keep_scheme: bool = False
+) -> Tuple[str, str]:
+    parsed = urlparse(uri)
+    if forced_scheme is not None:
+        scheme = "unix" if "unix" in forced_scheme.lower() else "http"
+    else:
+        scheme = parsed.scheme if parsed.scheme != "" else "http"
+
+    if scheme == "http" and parsed.scheme == "https":
+        scheme = "https"
+
+    endpoint = (
+        parsed.netloc + parsed.path
+    )  # To support with or without scheme, and for paths for unix
+
+    if "unix" in scheme:
+        # gRPC supports unix:path and the path is then relative, if the scheme is unix://, then the path is absolute
+        if endpoint.startswith("/"):
+            endpoint = "unix://" + endpoint
+        else:
+            endpoint = "unix:" + endpoint
+    elif keep_scheme:
+        endpoint = parsed.scheme + "://" + endpoint
+    return scheme, endpoint
```

Comment:
I'm not sure to follow the logic. Why do you need this?

Reply:
So it can be used elsewhere (notably in the worker creation) because depending on the context, the expected endpoint is either the netloc+path, or scheme+netloc+path